datafusion/physical-plan/src/topk/mod.rs

```diff
@@ -202,27 +205,116 @@ impl TopK {
             })
             .collect::<Result<Vec<_>>>()?;
 
+        // Selected indices in the input batch.
+        // Some indices may be pre-filtered if they exceed the heap’s current max value.
+
+        let mut selected_rows = None;
+
+        // If the heap doesn't have k elements yet, we can't create thresholds
+        if let Some(max_row) = self.heap.max() {
+            // Get the batch that contains the max row
+            let batch_entry = match self.heap.store.get(max_row.batch_id) {
+                Some(entry) => entry,
+                None => return internal_err!("Invalid batch ID in TopKRow"),
+            };
```

Comment:
I think you can write this a bit more succinctly

```suggestion
            let Some(entry) = self.heap.store.get(max_row.batch_id) else {
                return internal_err!("Invalid batch ID in TopKRow"),
            };
```